pkg/rewards/claim.go

```diff
@@ -75,75 +78,145 @@ func getClaimFlags() []cli.Flag {
+	p utils.Prompter,
+) error {
 	ctx := cCtx.Context
 	logger := common.GetLogger(cCtx)
 
-	config, err := readAndValidateClaimConfig(cCtx, logger)
+	yamlFile, err := os.ReadFile(config.BatchClaimFile)
 	if err != nil {
-		return eigenSdkUtils.WrapError("failed to read and validate claim config", err)
+		return eigenSdkUtils.WrapError("failed to read YAML config file", err)
 	}
-	cCtx.App.Metadata["network"] = config.ChainID.String()
 
-	ethClient, err := ethclient.Dial(config.RPCUrl)
-	if err != nil {
-		return eigenSdkUtils.WrapError("failed to create new eth client", err)
+	var claimConfigs []struct {
+		EarnerAddress  string   `yaml:"earner_address"`
+		TokenAddresses []string `yaml:"token_addresses"`
 	}
 
-	elReader, err := elcontracts.NewReaderFromConfig(
-		elcontracts.Config{
-			RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
-		},
-		ethClient, logger,
-	)
+	err = yaml.Unmarshal(yamlFile, &claimConfigs)
 	if err != nil {
-		return eigenSdkUtils.WrapError("failed to create new reader from config", err)
+		return eigenSdkUtils.WrapError("failed to parse YAML config", err)
 	}
 
-	df := httpProofDataFetcher.NewHttpProofDataFetcher(
-		config.ProofStoreBaseURL,
-		config.Environment,
-		config.Network,
-		http.DefaultClient,
-	)
+	var elClaims []rewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim
+
+	for _, claimConfig := range claimConfigs {
+		earnerAddr := gethcommon.HexToAddress(claimConfig.EarnerAddress)
+
+		var tokenAddrs []gethcommon.Address
+
+		// Empty token addresses list will create a claim for all tokens claimable
+		// by the earner address.
+		if len(claimConfig.TokenAddresses) == 0 {
+			tokenAddrs = nil
+		} else {
+			for _, addr := range claimConfig.TokenAddresses {
+				tokenAddrs = append(tokenAddrs, gethcommon.HexToAddress(addr))
+			}
+		}
+
+		elClaim, _, _, err := generateClaimPayload(
+			ctx,
+			config.ClaimTimestamp,
+			elReader,
+			logger,
+			earnerAddr,
+			df,
+			tokenAddrs,
+		)
+
+		elClaims = append(elClaims, *elClaim)
+
+		if err != nil {
+			logger.Errorf("Failed to process claim for earner %s: %v", earnerAddr.String(), err)
+			continue
+		}
+	}
+
+	if config.Broadcast {
+		eLWriter, err := common.GetELWriter(
+			config.ClaimerAddress,
+			config.SignerConfig,
+			ethClient,
+			elcontracts.Config{
+				RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
+			},
+			p,
+			config.ChainID,
+			logger,
+		)
+
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to get EL writer", err)
+		}
+
+		logger.Infof("Broadcasting batch claim transaction...")
+		receipt, err := eLWriter.ProcessClaims(ctx, elClaims, config.RecipientAddress, true)
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to process claim", err)
+		}
+
+		logger.Infof("Claim transaction submitted successfully")
+		common.PrintTransactionInfo(receipt.TxHash.String(), config.ChainID)
+	} else {
+		return eigenSdkUtils.WrapError("Batch claim currently only supported with broadcast flag", err)
```

Comment:
why? it should support non broadcast where you can see the unigned tx or pretty print batch claim. a lot of people use `calldata` for claiming

Reply:
I know previously operators were piping these claims to other CLI tools. What format do you think would work well here when outputting multiple claims for multiple earners?

Reply:
even tho it's multiple claims the calldata is only one right? function call is `ProcessClaims` which is just another smart contract call with array of claims. so doesn't matter how many earners are there.

Reply:
Confirmed it is just one calldata returned for multiple claims, I will add this in. I had originally thought we would need to output one calldata per earner.

Reply:
I refactored the output code into a `broadcastClaims` function and have initial support in here now for outputting the calldata and multiple claims. I want to tune it more once the contracts are live, likely we will want to add prefix for each earner address before claim data output.